src/common/src/hash/key.rs

```diff
@@ -42,6 +42,61 @@ use crate::util::hash_util::Crc32FastBuilder;
 use crate::util::iter_util::ZipEqFast;
 use crate::util::value_encoding::{deserialize_datum, serialize_datum_into};
 
+pub static MAX_GROUP_KEYS: usize = 64;
+
+/// Bitmap for null values in key.
+/// This is specialized for key,
+/// since it usually has few group keys.
+#[repr(transparent)]
+#[derive(Clone, Debug, PartialEq)]
+pub struct NullBitmap {
+    inner: Set64,
+}
+
+impl NullBitmap {
+    fn empty() -> Self {
+        NullBitmap {
+            inner: Set64::empty(),
+        }
+    }
+
+    fn is_empty(&self) -> bool {
+        self.inner.is_empty()
+    }
+
+    fn set_true(&mut self, idx: usize) {
+        self.inner.add_inplace(idx);
+    }
+
+    fn estimated_heap_size(&self) -> usize {
```

Comment:
I guess this should be zero as it's entirely on the stack. BTW, we can `impl EstimateSize` for this method.

Reply:
What if the struct is allocated to heap by user? Do we have to create a newtype for it, implementing EstimateSize with heap space?

Reply:
I guess so.